staging/src/k8s.io/apiserver/pkg/admission/options/authentication.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"k8s.io/client-go/rest"
+)
+
+// AuthenticationInfoResolverWrapper is a wrapper for AuthenticationInfoResolver.
```

Comment:
I don't understand what this type is needed for. Can you make the comment say something say something useful? Like, what can I do with this, why would I need it?

Reply:
Sorry, I was being lazy. Changed to
```
// AuthenticationInfoResolverWrapper can be used to inject Dial function to the
// rest.Config generated by the resolver.
```

It's used at https://github.com/kubernetes/kubernetes/blob/2d64ce5e8e45e26b02492d2b6c85e5ebfb1e4761/cmd/kube-apiserver/app/server.go#L449-L463

Maybe we should refactor the initializer function to `SetDial`, but that will be another PR.